journals/apps/api/v1/views.py

```diff
@@ -62,3 +67,20 @@ def _revoke_access(self, order_number):
                     access_record.journal,
                     access_record.order_number)
         return HttpResponse()
+
+
+class UserPageVisitView(generics.ListCreateAPIView):
```

Comment:
why are we using generics.ListCreateAPIView instead of viewsets.ModelViewSet that we use in other places?

Reply:
I tried `viewsets.ModelViewSet` but it is not suitable for us because in our case get/retrieve will returns more than one object because we are filtering over user_id, so to resolve this we have to change the default function of retirieve. That is why `generics.ListCreateAPIView` is more suitable for this specific case scenario. I am tagging you above where I used different apporach.

Reply:
not sure if that is true. pretty sure you can override get_queryset() in ModelViewSet as well to return whatever you want.

Reply:
@bill-filler you are right get_queryset() is responsible for data but in `list` method not in `GET/retrieve` method.

```
def retrieve(self, request, *args, **kwargs):
    instance = self.get_object()
    serializer = self.get_serializer(instance)
    return Response(serializer.data)
```
This function is responsible for `GET` request in `ModelViewSet`. We have to override this function to return the multiple records on `GET` request. In my earlier commits I tired this approch but then @ziafazal has pointed out that `generics.ListCreateAPIView` is more suitable for our case scenario.